### ISSUE TITLE
Accessing Spacehip module in an imported Fastfile (uninitialized constant error)

### ISSUE THREAD

**Issue by Skoti** [fastlane/fastlane#14678]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
In a shared _Fastfile_ I have a lane that is accessing `Spaceship::` module and makes use of some methods from it.
It all works when I import this _Fastfile_ locally through `import "path_to_fastfile"`
but when I use `import_from_git` action (with the same file) then I get `uninitialized constant` error wherever I access Spaceship.

To solve this I need to add `require "spaceship"` in that shared _Fastfile_ and it will work in both cases.

This is weird, what is the difference between those two imports? And why only one of them needs to require spaceship?
<br/>
Maybe this is somehow connected.
Actually, it all started a bit earlier by removing `fastlane_require "spaceship"` line from that shared _Fastfile_. I have had this line for a long time and it worked in both import cases, but I needed to remove it after upgrading to fastlane 2.121.1 because it tried to install `spaceship 1.0.0` gem 😮 -> and it was not an issue in previous fastlane releases as I clearly didn't have this gem installed (I have removed it immediately so it is not causing any other conflicts).
I am not sure if this was an issue in 2.121.0 because I might not manage to use it before upgrading to 2.121.1, but I can check it if this is desired.
<br/>
This can be easily reproduced with having just these lines below in a shared _Fastfile_.
Import the _Fastfile_ below via `import` and `import_from_git` to see differences.
```
# uncomment below to make it work for import_from_git action
# require "spaceship"
# or
# uncomment below to make it install spaceship 1.00 gem
# fastlane_require "spaceship"

lane :bu do |options|
  puts Spaceship::Tunes::AppStatus::PENDING_DEVELOPER_RELEASE
end
```

I am happy enough with this "require" workaround but I have a feeling that if this is working differently after an upgrade then something has changed (probably unintentionally) and it might be an underlying issue which might cause some other side effects.

**Comment by janpio** [comment]
Whoah, that's a strange one.

What version did you upgrade from - 2.120.0?

(We had other strangeness with `spaceship` and a missing import: https://github.com/fastlane/fastlane/issues/14661 Somehow this feels in the same ballpark)

**Comment by Skoti** [comment]
Yes, I have upgraded from _2.120.0_.
Although, I cannot reproduce the first case today - not sure why, might be due to a different session.
I have also checked that the _fastlane_require_ was still trying to install spaceship gem on previous fastlane versions like 2.114 and 2.119 - maybe I had it installed and didn't notice? 😞 
But it doesn't explain why it tried to install this gem after upgrading to 2.121.1 - if it worked before it would mean it was already installed 😮 and the only change in gems was this upgrade...

Anyway, seems like the other thread found the solution so I think it is ok to close this issue. 👍  Excuse me for all the confusion.
